libsolidity/analysis/ControlFlowGraph.h

```diff
@@ -0,0 +1,140 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <libsolidity/ast/AST.h>
+#include <libsolidity/ast/ASTVisitor.h>
+#include <libsolidity/interface/ErrorReporter.h>
+
+#include <map>
+#include <memory>
+#include <stack>
+#include <vector>
+
+namespace dev
+{
+namespace solidity
+{
+
+/// Basic Control Flow Block.
+/// Basic block of control flow. Consists of a set of AST nodes
+/// for which control flow is always linear.
+struct ControlFlowBlock
+{
+	/// All variable declarations executed in this control flow block.
+	std::vector<VariableDeclaration const*> variableDeclarations;
+	/// All expressions executed in this control flow block (this includes all subexpressions!).
```

Comment:
`evaluated` instead of `executed`?

Reply:
I could specify the current order of these vectors, but IMO order does not and should not matter for a control flow graph. It's not an AST - the only question the CFG wants to answer is whether something is present in a path of control flow or not. I documented them as "unordered" now.